### ISSUE TITLE
0.27.0-beta* fails to compile with tokio-rustls-tls

### ISSUE THREAD

**Issue by jplatte** [durch/rust-s3#149]
**Describe the bug**

Trying to use 0.27.0-beta{1,2,4,5} with `default-features = false, features = ["tokio-rustls-tls"]` fails.

**To Reproduce**

Create a crate with a dependency on the beta version, like this:

```toml
rust-s3 = { version = "=0.27.0-beta5", default-features = false, features = ["tokio-rustls-tls"] }
```

**Expected behavior**

It should work

**Environment**
 - Rust version: rustc 1.48.0 (7eac88abb 2020-11-16)
 - lib version 0.27.0-beta5

**Additional context**

Full error message:

```
    Checking rust-s3 v0.27.0-beta5
error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:103:23
    |
103 |         let request = RequestImpl::new(self, path.as_ref(), Command::PresignGet { expiry_secs });
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:138:23
    |
138 |         let request = RequestImpl::new(
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:191:23
    |
191 |         let request = RequestImpl::new(&bucket, "", command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:244:23
    |
244 |         let request = RequestImpl::new(&bucket, "", command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:287:23
    |
287 |         let request = RequestImpl::new(self, "", command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:425:23
    |
425 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:473:23
    |
473 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:518:23
    |
518 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:577:23
    |
577 |         let request = RequestImpl::new(self, &s3_path, command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:593:41
    |
593 |                     let abort_request = RequestImpl::new(self, &path, abort);
    |                                         ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:605:35
    |
605 |                     let request = RequestImpl::new(self, &path, command);
    |                                   ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:623:44
    |
623 |                     let complete_request = RequestImpl::new(self, &path, complete);
    |                                            ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:636:31
    |
636 |                 let request = RequestImpl::new(self, &path, command);
    |                               ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:751:23
    |
751 |         let request = RequestImpl::new(self, "?location", Command::GetBucketLocation);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:811:23
    |
811 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:850:23
    |
850 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:901:23
    |
901 |         let request = RequestImpl::new(self, path.as_ref(), command);
    |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
    --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:1009:23
     |
1009 |         let request = RequestImpl::new(self, path, command);
     |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
    --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:1048:23
     |
1048 |         let request = RequestImpl::new(self, path.as_ref(), command);
     |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
    --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:1090:23
     |
1090 |         let request = RequestImpl::new(self, path.as_ref(), command);
     |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0433]: failed to resolve: use of undeclared type `RequestImpl`
    --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:1120:23
     |
1120 |         let request = RequestImpl::new(self, "/", command);
     |                       ^^^^^^^^^^^ use of undeclared type `RequestImpl`

error[E0412]: cannot find type `Path` in this scope
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:563:26
    |
563 |         path: impl AsRef<Path>,
    |                          ^^^^ not found in this scope
    |
help: consider importing one of these items
    |
3   | use crate::bucket::serde_xml::ser::ser::Path;
    |
3   | use std::path::Path;
    |

error[E0433]: failed to resolve: use of undeclared type `File`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:566:38
    |
566 |         self._put_object_stream(&mut File::open(path).await?, s3_path.as_ref())
    |                                      ^^^^ not found in this scope
    |
help: consider importing one of these items
    |
3   | use crate::bucket::serde_xml::ser::ser::net::io::sys::process::process_common::File;
    |
3   | use crate::bucket::serde_xml::ser::ser::net::io::sys::rand::imp::File;
    |
3   | use std::fs::File;
    |

error[E0405]: cannot find trait `AsyncRead` in this scope
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:571:36
    |
571 |     async fn _put_object_stream<R: AsyncRead + Unpin + Send>(
    |                                    ^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `read_chunk` in module `crate::utils`
   --> /home/jplatte/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-s3-0.27.0-beta5/src/bucket.rs:587:39
    |
587 |             let chunk = crate::utils::read_chunk(reader).await?;
    |                                       ^^^^^^^^^^ not found in `crate::utils`

error: aborting due to 25 previous errors

Some errors have detailed explanations: E0405, E0412, E0425, E0433.
For more information about an error, try `rustc --explain E0405`.
error: could not compile `rust-s3`
```

**Comment by durch** [comment]
@jplatte docs, are not great here yet, you also need to include the `with-tokio` feature to get this to work

**Comment by jplatte** [comment]
Why does the tokio-rustls-tls feature not activate that one?

**Comment by durch** [comment]
There is no good reason, I will push out an update to streamline features a bit

**Comment by durch** [comment]
Fixed in `0.27.0-beta6`
